resources/scenarios/signet_miner.py

```diff
@@ -0,0 +1,568 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+###
+# WARNET SCENARIO SIGNET MINER
+#
+# This file is a hacked-up version of Bitcoin Core contrib/signet/miner.
+# The primary difference is that instead of using bitcoin-cli for RPCs
+# we use the authproxy from the test framework.
+###
+
+# The base class exists inside the commander container
+try:
+    from commander import Commander
+except ImportError:
+    from resources.scenarios.commander import Commander
+
+import json
+import logging
+import math
+import re
+import struct
+import sys
+import time
+import subprocess
+
+from test_framework.blocktools import get_witness_script, script_BIP34_coinbase_height # noqa: E402
+from test_framework.messages import CBlock, CBlockHeader, COutPoint, CTransaction, CTxIn, CTxInWitness, CTxOut, from_binary, from_hex, ser_string, ser_uint256, tx_from_hex # noqa: E402
+from test_framework.psbt import PSBT, PSBTMap, PSBT_GLOBAL_UNSIGNED_TX, PSBT_IN_FINAL_SCRIPTSIG, PSBT_IN_FINAL_SCRIPTWITNESS, PSBT_IN_NON_WITNESS_UTXO, PSBT_IN_SIGHASH_TYPE # noqa: E402
+from test_framework.script import CScriptOp # noqa: E402
+
+logging.basicConfig(
+    format='%(asctime)s %(levelname)s %(message)s',
+    level=logging.INFO,
+    datefmt='%Y-%m-%d %H:%M:%S')
+
+SIGNET_HEADER = b"\xec\xc7\xda\xa2"
+PSBT_SIGNET_BLOCK = b"\xfc\x06signetb"    # proprietary PSBT global field holding the block being signed
+RE_MULTIMINER = re.compile(r"^(\d+)(-(\d+))?/(\d+)$")
```

Comment:
iirc, this puts us on our own "private signet," correct?

Reply:
no the scenario is is a copy/paste from bitcoin core with a few changes to redirect the RPCs. The "private signet" is configured by the `-signetchallenge` setting, which I believe also sets a custom magic bytes. Not sure if that's the same as the SIGNET_HEADER here